src/Facebook/FacebookClient.php

```diff
@@ -203,32 +193,29 @@ public function sendRequest(FacebookRequest $request)
 
         list($url, $method, $headers, $body) = $this->prepareRequestMessage($request);
 
-        // Since file uploads can take a while, we need to give more time for uploads
-        $timeOut = static::DEFAULT_REQUEST_TIMEOUT;
-        if ($request->containsFileUploads()) {
-            $timeOut = static::DEFAULT_FILE_UPLOAD_REQUEST_TIMEOUT;
-        } elseif ($request->containsVideoUploads()) {
-            $timeOut = static::DEFAULT_VIDEO_UPLOAD_REQUEST_TIMEOUT;
-        }
-
-        // Should throw `FacebookSDKException` exception on HTTP client error.
-        // Don't catch to allow it to bubble up.
-        $rawResponse = $this->httpClientHandler->send($url, $method, $body, $headers, $timeOut);
+        $psr7Request = MessageFactoryDiscovery::find()->createRequest($method, $url, $headers, $body);
```

Comment:
So you suggest something like:
```php
$psr7Response = $this->httpClient->sendRequest(MessageFactoryDiscovery::find()->createRequest($method, $url, $headers, $body););
```
If that's the case, I think I prefer how it is now, it makes it clear that we send PSR7 request (and avoid a long line) :)